packages/yarnpkg-fslib/sources/ZipFS.ts

```diff
@@ -1099,9 +1114,10 @@ export class ZipFS extends BasePortableFakeFS {
     this.registerEntry(resolvedP, index);
 
     const rc = this.libzip.file.setExternalAttributes(this.zip, index, 0, 0, this.libzip.ZIP_OPSYS_UNIX, (0o120000 | 0o777) << 16);
-    if (rc === -1) {
+    if (rc === -1)
       throw new Error(this.libzip.error.strerror(this.libzip.getError(this.zip)));
-    }
+
+    this.symlinkCount += 1;
```

Comment:
Shouldn't `unlinkSync` decrease this refcount?

Reply:
Yes, but since all that function does is throw an error I left it alone.
https://github.com/yarnpkg/berry/blob/3a7e1df175e37a6397c9c87b70300506726aeef7/packages/yarnpkg-fslib/sources/ZipFS.ts#L1012-L1014

Reply:
Oops, indeed 😄